### PR TITLE
Add content type snippet unit tests

### PR DESCRIPTION
**Pull request by gormal**
### Motivation

Add unit tests for content type snippets.

Two breaking changes:
CreateContentSnippetCreateModel => ContentTypeSnippetCreateModel 
SnippetPatchRemoveModel => ContentTypeSnippetPatchRemoveModel

### CODE REVIEW

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        {
+            var client = _fileSystemFixture.CreateMockClientWithResponse("Snippet.json");
+
+            var expected = _fileSystemFixture.GetExpectedResponse<ContentTypeSnippetModel>("Snippet.json");
+
+            await client.Invoking(x => x.ModifyContentTypeSnippetAsync(Reference.ByCodename("metadata"), null)).Should().ThrowAsync<ArgumentException>();
+        }
+
+        [Fact]
+        public async void ModifyContentTypeSnippetAsync_NoChanges_ModifiesContentType()
```
Comment (Sevitas) [inline Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs:167]: test name is _ModifyContentTypeSnippetAsync_NoChanges_ModifiesContentType_ but actually checks throwing an exception. There are also other places with the same issue.

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+            response.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public async void ModifyContentTypeSnippetAsync_IdentifierIsNull_ModifiesContentType()
+        {
+            var client = _fileSystemFixture.CreateMockClientWithResponse("Snippet.json");
+
+            var expected = _fileSystemFixture.GetExpectedResponse<ContentTypeSnippetModel>("Snippet.json");
```
Comment (Sevitas) [inline Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs:144]:
```suggestion
            _fileSystemFixture.GetExpectedResponse<ContentTypeSnippetModel>("Snippet.json");
```

**Review by Sevitas — CHANGES_REQUESTED** [review]

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs (hunk 1 above)
Comment (gormal) [reply Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/ContentTypeSnippetTests.cs]: There were also naming problems in ContentType tests. Fixed it all.

**Review by Sevitas — APPROVED** [review]
LGTM!

**Pull request merged**